units/systemd-vconsole-setup.service.in

```diff
@@ -7,6 +7,13 @@
 #  the Free Software Foundation; either version 2.1 of the License, or
 #  (at your option) any later version.
 
+{% if ENABLE_FIRSTBOOT %}
+# Note that this service exists in two copies: systemd-vconsole-setup.service
+# and systemd-vconsole-setup-reload.service. The second one is used internally
+# by systemd-firstboot.service to start a second copy of the unit without
+# ordering dependencies. The first copy should be used in all other cases.
+{% endif %}
```

Comment:
hmm, so I understand this is purely about concurrently issuing the TTY reset operation and initializing the VT, right?

so what about simply taking a BSD file lock on /dev/console right before doing the two operations and releasing it right-after? that sounds pretty robust, no, and isn't hacky, at all?

Reply:
Hmm, but if we take a lock, we'd also need the other side to do the same, i.e. we'd need to block pid1 on the lock so that it doesn't reset the tty underneath us while it's preparing to launch `firstboot.service`. That doesn't seem right.

Reply:
Yeah I checked and its the child process that resets the fd, so it should be fine to block there.

I wonder if this would be a better solution in general instead of adding all the ordering between services. We can just BSD lock the tty every time we do some interruptive stuff in terminal-util.c, and BSD lock the tty while we execute loadkeys in vconsole-setup.service.

Although ordering the getty units after systemd-vconsole-setup.service makes sense either way I think. But it means we wouldn't have to order every service using vhangup after systemd-vconsole-setup.service anymore.